File: src/models/notebook-observation-kind.ts
export const NotebookObservationKind = {
  /** @original `OVS_DefautCarnet` */
  LogBookIssue: 0,

  /** @original `OVS_ObservationParent` */
  Observation: 1,

  /** @original `OVS_Encouragement` */
  Encouragement: 2,

  /** @original `OVS_Autres` */
  Other: 3
} as const;

export type NotebookObservationKind = typeof NotebookObservationKind[keyof typeof NotebookObservationKind];

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/NotebookObservationKind.kt
package ink.literate.pawnote.models

enum class NotebookObservationKind(val code: Int) {
  LogBookIssue(0),
  Observation(1),
  Encouragement(2),
  Other(3);

  companion object {
    fun fromInt(code: Int) = entries.first { it.code == code }
  }
}
